### PR TITLE
Improve UX of dialog buttons

### PR DESCRIPTION
**Pull request by LMS007**
- Improve UX of dialog buttons and make them more consistent with both the sidebar and the grader
- Solve issue where it was impossible to know if the submit button was disabled in the file picker

-------

In solving the original ticket. I thought initially it would be best to add the `required` attribute to the file picker, but that was not really the right solution as the UI was fundamentally different. The URL picker is just a field missing and the other was a selection of items from a list and the native `required` attribute would not work the same (or as plug-and-play'ish). Furthermore it does make sense to disable the button while things are loading and I think we should keep that behavoir.  _I think a follow up could be to use the client validator and replace the native required UX for the URLPicker as it does not really match much of anything currently._ 

The issues that became more apparent were how we were 1. dealing with the styling of a disabled button and 2. these dialogs are not at all consistent with the any other of our buttons in the sidebar or grader and I can't recall any reasoning for this? 

In any case, This changes the behavior to make buttons darker on hover (like our sidebar and grader) and removes the hover effect for disabled buttons (like our sidebar and grader). Note: WCAG compatibility does not have special requirement either since its decorative in case anyone was wondering.

https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html#:~:text=Incidental%3A%20Text%20or%20images%20of,content%2C%20have%20no%20contrast%20requirement.

Furthermore, the brand color (~red) seems to only be used for activated menu buttons in our sidebar e.g. (which I do like)

Screen Shot 2020-12-09 at 1 25 20 PM

But Since we really didn't have those cases in LMS, I just remove that styling for now.

dialog_buttons

Fixes https://github.com/hypothesis/lms/issues/2280

### CODE REVIEW

File: lms/static/styles/_variables.scss
Comment (LMS007) [inline lms/static/styles/_variables.scss:13]: Pulled this terrible color in form the sidebar, It really need it  unfortunately  since it is the button color over there and its just too light if we use $grey-5.

File: lms/static/styles/components/_Button.scss
Comment (LMS007) [inline lms/static/styles/components/_Button.scss:20]: This is how we do disabled styling in the sidebar

File: lms/static/styles/components/_Dialog.scss
Comment (LMS007) [inline lms/static/styles/components/_Dialog.scss:52]: beefed up the "X" button, It just looked too light to me and since the hover-color is no longer red, it needs more weight to be able to see that effect.

File: lms/static/styles/components/_Button.scss
Comment (LMS007) [inline lms/static/styles/components/_Button.scss:46]: This was unused css.

File: lms/static/styles/_variables.scss
Comment (esanzgar) [reply lms/static/styles/_variables.scss:13]: Some alternative:

```suggestion
$grey-5\.5: #595959;
```

```suggestion
$grey-5-5: #595959;
```

```suggestion
$grey-5-tmp: #595959;
```

Comment (esanzgar) [reply lms/static/styles/_variables.scss:13]: Why is this an interim color? Why don't we fully adopt it?

If we would adopt this new grey value, it may be makes sense to convert all:
*  `grey-7` to `grey-8`
* `grey-6` to `grey-7`

**Review by esanzgar — APPROVED** [review]
It is challenging to work with a monochromatic grey palette!

I was taking a look at [google's material design](https://material.io/components/buttons#hierarchy-and-placement) and they have a few good ideas that we could adopt. It maybe worth discussing these.

**Comment by seanh** [comment]
This makes all the previously black buttons throughout the app grey instead (and they only turn fully black on hover), was that intentional?

Currently on master:

Screenshot from 2020-12-11 14-35-24

On this branch:

Screenshot from 2020-12-11 14-34-54
Screenshot from 2020-12-11 14-36-59
Screenshot from 2020-12-11 14-36-19
Screenshot from 2020-12-11 14-35-57

**Comment by robertknight** [comment]
Hello - Just a heads up. To ensure consistency across the client and LMS frontend I think we should review this collectively. In particular we should get input from @lyzadanger who has done much of the related work on the client side. We also have a number of other button-related issues we need to discuss.

**Review by robertknight — CHANGES_REQUESTED** [review]
I haven't had a chance to look at this in detail yet, but can we hold off merging this for the moment per my previous comment. There was also a related Slack thread here: https://hypothes-is.slack.com/archives/C1M8NH76X/p1607695129241700.

**Comment by LMS007** [comment]
> This makes all the previously black buttons throughout the app grey instead (and they only turn fully black on hover), was that intentional?


Yes, you're exactly correct and this also matches the style in the sidebar and grader. It has been changed for consistency and I did not question validity. Perhaps we should at this juncture? We have a lot of gray and it might serve us well to have some colors here and there.

File: lms/static/styles/_variables.scss
Comment (LMS007) [reply lms/static/styles/_variables.scss:13]: I think we do revisit this rule next week. It was dropped in to the client as  a shim many moons ago, but we did not expect it to be long lived. We were at that time in the process of developing a new color palette but that was put on hold so its long outlived it expected life span.

We should keep the name the same as the client (if we do end up using it) OR move to a new pallet entirely and then abandon it.

**Pull request closed without merging**
